### PR TITLE
Fix S3 error handling

### PR DESCRIPTION
**Pull request by rullzer**

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]
Seems to do the trick when manually triggering an exception :+1:

File: lib/private/Files/ObjectStore/ObjectStoreStorage.php [hunk 1]
```diff
@@ -461,19 +467,33 @@ public function writeStream(string $path, $stream, int $size = null): int {
...
+					'app' => 'objectstore',
+					'message' => 'Could not update object ' . $urn . ' for ' . $path,
+				]);
+			}
 			throw $ex; // make this bubble up
 		}
 
-		if (!$exists) {
+		if ($exists) {
+			$this->getCache()->update($fileId, $stat);
```
Comment (icewind1991) [inline lib/private/Files/ObjectStore/ObjectStoreStorage.php:495]: what stats are being updated here? the size is already updated in the `CountWrapper` callback

Comment (rullzer) [reply lib/private/Files/ObjectStore/ObjectStoreStorage.php:495]: Mtime etc

**Review by icewind1991 — APPROVED** [review]

**Pull request merged**